src/core/Akka.Tests/Actor/CoordinatedShutdownSpec.cs

```diff
@@ -411,5 +411,60 @@ public void CoordinatedShutdown_must_terminate_ActorSystem()
             Sys.WhenTerminated.IsCompleted.Should().BeTrue();
             CoordinatedShutdown.Get(Sys).ShutdownReason.ShouldBeEquivalentTo(customReason);
         }
+
+        [Fact]
+        public async Task CoordinatedShutdown_must_be_run_by_ActorSystem_Terminate()
+        {
+            await Sys.Terminate();
+            Sys.WhenTerminated.IsCompleted.Should().BeTrue();
+            CoordinatedShutdown.Get(Sys).ShutdownReason.ShouldBeEquivalentTo(CoordinatedShutdown.ActorSystemTerminateReason.Instance);
+        }
+
+        [Fact]
+        public async Task CoordinatedShutdown_must_not_be_run_by_ActorSystem_Terminate_when_run_by_actor_system_terminate_is_off()
+        {
+            var sys = ActorSystem.Create(
+                "name", 
+                ConfigurationFactory
+                    .ParseString("akka.coordinated-shutdown.run-by-actor-system-terminate = off")
+                    .WithFallback(Sys.Settings.Config));
+            var actor = CoordinatedShutdown.Get(sys);
+
+            try
+            {
+                await sys.Terminate();
+                sys.WhenTerminated.IsCompleted.Should().BeTrue();
+                actor.ShutdownReason.ShouldBeEquivalentTo(null);
+            }
+            finally
+            {
+                Shutdown(sys);
+            }
+        }
+
+        [Fact]
+        public void CoordinatedShutdown_must_not_allow_terminate_actor_system_set_to_off_and_run_by_actor_system_terminate_set_to_on()
+        {
+            Action act = () => {
+                ActorSystem sys = null;
+                try
+                {
+                    sys = ActorSystem.Create(
+                        "name",
+                            ConfigurationFactory
+                                .ParseString("akka.coordinated-shutdown.terminate-actor-system = off")
```

Comment:
Just to make this test easier to understand for future maintainers, we should explicitly include both conflicting parts of the HOCON configuration here, rather than relying on the base `Config` used in this spec.

Reply:
Done